Apps/Src/UpdateVelocity.c

```diff
@@ -50,14 +59,29 @@ void Task_UpdateVelocity(void *p_arg)
         // Deadband comparison
         if(brakePedalPercent >= UNTOUCH_PEDALS_PERCENT){
             desiredVelocity = 0;
-            desiredMotorCurrent = 0;
-        } else {
-            if (Switches_Read(FOR_SW)) {
+            desiredMotorCurrent = Switches_Read(REGEN_SW) ? REGEN_CURRENT : 0;
+        } 
+        else if(Switches_Read(REGEN_SW) 
+                && (MotorController_ReadVelocity() > THRESHOLD_VEL) 
+                && RegenEnable
+                && accelPedalPercent < NEUTRAL_PEDALS_PERCENT){ 
+            
+            desiredVelocity = 0;
+            // set regen current based on how much the accel pedal is pressed down
+            desiredMotorCurrent = REGEN_CURRENT * REGEN_RANGE / (NEUTRAL_PEDALS_PERCENT - accelPedalPercent);
```

Comment:
Won't this regen more when the accelerator is pushed closer to Neutral point? Ie. if accelPedalPercent is 24, desired motorCurrent will be more than if accelPedalPercent is 10. Wouldn't we want higher regen current if the accelerator is pushed less?

(just a quick thought on how to implement regen getting stronger as the pedal is pushed less)
```suggestion
            #define REGEN_CURVE_OFFSET = 25 // offset to translate the braking strength curve to the right, shouldn't exceed 25, or we might get a negative denominator. Should be used to adjust the steepness of the regen curve here.
            desiredMotorCurrent = REGEN_CURRENT * REGEN_RANGE / (NEUTRAL_PEDALS_PERCENT +  accelPedalPercent-REGEN_STRENGTH_OFFSET : 0);
```

Reply:
Yes, that's correct. I'll fix this